x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -543,7 +548,7 @@ export class RulesClient {
 
     // default duration of instance summary is 60 * rule interval
     const dateNow = new Date();
-    const durationMillis = parseDuration(rule.schedule.interval) * 60;
+    const durationMillis = parseDuration(rule.schedule.interval) * (numberOfExecutions ?? 60);
```

Comment:
Due to task manager drift, we often don't see the exact number of executions inside the time interval calculated from numExecutions * schedule. For a rule running every minute, querying for the last 60 minutes might retrieve up to 60 executions (most likely 50-59). Would it make sense to add some padding to this to try to ensure we actually get the number of executions we want?

The other option would be to add a second query for `execute` events from the event log with the size param set to `numberOfExecutions`. That would provide the most exact results.

Reply:
Great point here ^^

Let's not query based on time but rather actual number of executions

Reply:
So the way I figured out how to do this was by passing to `findEventsBySavedObjectIds` the following:
```js
     {
          page: 1,
          per_page: numberOfExecutions * 2,
          end: dateNow.toISOString(),
          sort_order: 'desc',
        },
```

I have no idea why I need to do `numberOfExecutions * 2`. Passing `numberOfExecutions` to `per_page` only retrieves half the actual executions that I'm looking for.

Anyone know what that's about? Is this acceptable or is there a bug?

Reply:
That's weird @Zacqary , definitely not supposed to work this way 🤔 
I'd definitely get to the bottom of it before merging anything. @pmuellr is the goto person for Event Log APIs, so I'd loop him in 👍

Reply:
I totally see where @ymao1 is coming from here and if this UX **is** long term, I'd definitely consider adding a second API call.

We have been considering replacing the alerts list with the alerts table in the future, but that would probably only apply to Alerts-as-data supported rule types. 🤔 So this UX **is** likely to live for a while.

What do you think @XavierM ?

Reply:
Yeah, I think without using the filter, you'll get all the events - you want to filter on 

    event.provider: alerting AND event.action:execute

Reply:
I talked to @ymao1 to have a better understanding here, you know newbie. if I understand everything correctly what we want here is to create a new query on `findEventsBySavedObjectIds` with the filter `event.provider: alerting AND event.action:execute` and `per_page: numberOfExecutions`. Then we will create a new function to only build the `excution_duration` attribute from the result of this new query.

```code
       if (event?.event?.duration) {
        const eventDirationMillis = event?.event?.duration / Millis2Nanos;
        eventDurations.push(eventDirationMillis);
        if (event?.['@timestamp']) {
          eventDurationsWithTimestamp[event?.['@timestamp']] = eventDirationMillis;
        }
      }
      ...
      if (eventDurations.length > 0) {
       alertSummary.executionDuration = {
         average: Math.round(mean(eventDurations)),
         valuesWithTimestamp: eventDurationsWithTimestamp,
       };
     }
```
https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/lib/alert_summary_from_event_log.ts#L62-L85
https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/lib/alert_summary_from_event_log.ts#L129-L134

Reply:
This could be done as an agg, right - the average of the durations?  There's some code here - [`x-pack/plugins/actions/server/usage/actions_telemetry.ts`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/actions/server/usage/actions_telemetry.ts), which is doing some elaborate searches over the event log, at a lower level than we'd want.  We intend on adding a search-with-aggs capability to the EL client, haven't had the time to merge it (PR https://github.com/elastic/kibana/pull/91731 needs to be updated).  

If you have time, would be great to get that aggs function in - there's plans for more telemetry which will likely require aggs - someone will hopefully get to it soon :-)